autosklearn/ensemble_builder.py

```diff
@@ -332,41 +332,37 @@ def main(self, return_pred=False):
         if return_pred:
             return valid_pred, test_pred
 
-    def get_worst_disk_consumption(self):
+    def get_disk_consumption(self, pred_path):
         """
         gets the cost of a model being on disc
         """
 
-        # Loop through the files currently in the directory
-        worst_consumption = 0
-        for pred_path in self.y_ens_files:
-
-            match = self.model_fn_re.search(pred_path)
-            _full_name = match.group(0)
-            _seed = match.group(1)
-            _num_run = match.group(2)
-            _budget = match.group(3)
-
-            # Besides the prediction, we have to take care of three other files: model,
-            # validation and test.
-            model_name = '%s.%s.%s.model' % (_seed, _num_run, _budget)
-            model_path = os.path.join(self.dir_models, model_name)
-            pred_valid_name = 'predictions_valid' + _full_name
-            pred_valid_path = os.path.join(self.dir_valid, pred_valid_name)
-            pred_test_name = 'predictions_test' + _full_name
-            pred_test_path = os.path.join(self.dir_test, pred_test_name)
-
-            paths = [model_path, pred_path]
-            if os.path.exists(pred_valid_path):
-                paths.append(pred_valid_path)
-            if os.path.exists(pred_test_path):
-                paths.append(pred_test_path)
-            this_model_cost = sum([os.path.getsize(path) for path in paths])
-            if this_model_cost > worst_consumption:
-                worst_consumption = this_model_cost
-
-        # get the worst consumption in megabytes
-        return round(worst_consumption / math.pow(1024, 2), 2)
+        match = self.model_fn_re.search(pred_path)
+        if not match:
+            raise ValueError("Invalid path format %s" % pred_path)
+        _full_name = match.group(0)
+        _seed = match.group(1)
+        _num_run = match.group(2)
+        _budget = match.group(3)
+
+        # Besides the prediction, we have to take care of three other files: model,
+        # validation and test.
+        model_name = '%s.%s.%s.model' % (_seed, _num_run, _budget)
+        model_path = os.path.join(self.dir_models, model_name)
+        pred_valid_name = 'predictions_valid' + _full_name
+        pred_valid_path = os.path.join(self.dir_valid, pred_valid_name)
+        pred_test_name = 'predictions_test' + _full_name
+        pred_test_path = os.path.join(self.dir_test, pred_test_name)
+
+        paths = [model_path, pred_path]
+        if os.path.exists(pred_valid_path):
+            paths.append(pred_valid_path)
+        if os.path.exists(pred_test_path):
+            paths.append(pred_test_path)
+        this_model_cost = sum([os.path.getsize(path) for path in paths])
+
+        # get the megabytes
+        return round(this_model_cost / math.pow(1024, 2), 2)
```

Comment:
Can this become zero? If yes, it's ambiguous with respect to the initial value of `self.read_preds[y_ens_fn]` and I suggest changing the initial value to -1 or None.

Reply:
None will be the default value. If we fail to read the data structure, the prediction will be ignored form the calculation.